### DISCUSSION TITLE
Error P3009

### DISCUSSION THREAD

**Discussion by pbmagnet4** [formbricks/formbricks#4341] [Self-hosting]
Hello,

I have been running Formbricks for sometime with great success but with this recent update, I have been getting the following error where I can no longer launch the application. Any advise on getting this to work would be greatly apprecaited.

[_Formbricks_logs (1).txt](https://github.com/user-attachments/files/17837049/_Formbricks_logs.1.txt)

**Comment by pbmagnet4** [comment]
migrate found failed migrations in the target database, new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
The `20241017124431_add_documents_and_insights` migration started at 2024-11-19 21:20:21.216341 UTC failed
npm error Lifecycle script `db:migrate:deploy` failed with error:
npm error code 1
npm error path /home/nextjs/packages/database
npm error workspace @formbricks/database@0.1.0
npm error location /home/nextjs/packages/database
npm error command failed
npm error command sh -c env DATABASE_URL="${MIGRATE_DATABASE_URL:-$DATABASE_URL}" prisma migrate deploy
> @formbricks/database@0.1.0 db:migrate:deploy
> env DATABASE_URL="${MIGRATE_DATABASE_URL:-$DATABASE_URL}" prisma migrate deploy
Prisma schema loaded from schema.prisma
Datasource "db": PostgreSQL database "formbricks", schema "public" at "formbricks-db:5432"
99 migrations found in prisma/migrations

Error: P3009
migrate found failed migrations in the target database, new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
The `20241017124431_add_documents_and_insights` migration started at 2024-11-19 21:20:21.216341 UTC failed
npm error Lifecycle script `db:migrate:deploy` failed with error:
npm error code 1
npm error path /home/nextjs/packages/database
npm error workspace @formbricks/database@0.1.0
npm error location /home/nextjs/packages/database
npm error command failed
npm error command sh -c env DATABASE_URL="${MIGRATE_DATABASE_URL:-$DATABASE_URL}" prisma migrate deploy
> @formbricks/database@0.1.0 db:migrate:deploy
> env DATABASE_URL="${MIGRATE_DATABASE_URL:-$DATABASE_URL}" prisma migrate deploy
Prisma schema loaded from schema.prisma
Datasource "db": PostgreSQL database "formbricks", schema "public" at "formbricks-db:5432"
99 migrations found in prisma/migrations

Error: P3009
migrate found failed migrations in the target database, new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
The `20241017124431_add_documents_and_insights` migration started at 2024-11-19 21:20:21.216341 UTC failed
npm error Lifecycle script `db:migrate:deploy` failed with error:
npm error code 1
npm error path /home/nextjs/packages/database
npm error workspace @formbricks/database@0.1.0
npm error location /home/nextjs/packages/database
npm error command failed
npm error command sh -c env DATABASE_URL="${MIGRATE_DATABASE_URL:-$DATABASE_URL}" prisma migrate deploy
> @formbricks/database@0.1.0 db:migrate:deploy
> env DATABASE_URL="${MIGRATE_DATABASE_URL:-$DATABASE_URL}" prisma migrate deploy
Prisma schema loaded from schema.prisma
Datasource "db": PostgreSQL database "formbricks", schema "public" at "formbricks-db:5432"
99 migrations found in prisma/migrations

Error: P3009
migrate found failed migrations in the target database, new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
The `20241017124431_add_documents_and_insights` migration started at 2024-11-19 21:20:21.216341 UTC failed
npm error Lifecycle script `db:migrate:deploy` failed with error:
npm error code 1
npm error path /home/nextjs/packages/database
npm error workspace @formbricks/database@0.1.0
npm error location /home/nextjs/packages/database
npm error command failed
npm error command sh -c env DATABASE_URL="${MIGRATE_DATABASE_URL:-$DATABASE_URL}" prisma migrate deploy
> @formbricks/database@0.1.0 db:migrate:deploy
> env DATABASE_URL="${MIGRATE_DATABASE_URL:-$DATABASE_URL}" prisma migrate deploy
Prisma schema loaded from schema.prisma
Datasource "db": PostgreSQL database "formbricks", schema "public" at "formbricks-db:5432"
99 migrations found in prisma/migrations

Error: P3009
migrate found failed migrations in the target database, new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
The `20241017124431_add_documents_and_insights` migration started at 2024-11-19 21:20:21.216341 UTC failed
npm error Lifecycle script `db:migrate:deploy` failed with error:
npm error code 1
npm error path /home/nextjs/packages/database
npm error workspace @formbricks/database@0.1.0
npm error location /home/nextjs/packages/database
npm error command failed
npm error command sh -c env DATABASE_URL="${MIGRATE_DATABASE_URL:-$DATABASE_URL}" prisma migrate deploy

**Comment by Saanvi26** [comment]
Any solutions for this? I have also been encountering this error.

**Comment by gupta-piyush19** [comment]
Hi @pbmagnet4, @Saanvi26 
We already have a discussion thread for this: https://github.com/formbricks/formbricks/discussions/4481
Feel free to ask questions if it doesn't work for you.
